lib/libzfs/libzfs_pool.c

```diff
@@ -3854,6 +3854,10 @@ zpool_history_unpack(char *buf, uint64_t bytes_read, uint64_t *leftover,
 		if (ISP2(*numrecords + 1)) {
 			*records = realloc(*records,
 			    *numrecords * 2 * sizeof (nvlist_t *));
+			if (*records == NULL) {
```

Comment:
@dinatale2 i agree that your variant is more safe, @behlendorf reviewed earlier `In the case of an error records must be considered undefined so you don't strictly need the tmp variable here`, that's why i removed it. We can just add comments for `zpool_history_unpack` that if there are any ENOMEM then we have to return ENOMEM to user. If i understand the code correctly, in such situation `records` won't be used.